File: classfields/cabinet-api-tests/api-client/src/main/java/ru/auto/tests/cabinet/provider/RequestSpecSupplier.java
package ru.auto.tests.cabinet.provider;

import io.restassured.builder.RequestSpecBuilder;
import io.restassured.config.LogConfig;
import io.restassured.config.RestAssuredConfig;
import ru.auto.tests.commons.restassured.AllureFailureListener;
import ru.auto.tests.commons.restassured.AllureLoggerFilter;

import java.net.URI;

import static io.restassured.config.FailureConfig.failureConfig;
import static io.restassured.config.LogConfig.logConfig;
import static io.restassured.config.ObjectMapperConfig.objectMapperConfig;
import static io.restassured.config.RestAssuredConfig.config;
import static io.restassured.http.ContentType.JSON;
import static ru.auto.tests.cabinet.GsonObjectMapper.gson;

class RequestSpecSupplier {

    private RequestSpecSupplier() {

    }
    static RequestSpecBuilder baseReqSpecBuilder(URI baseUri, String basePath) {
        return new RequestSpecBuilder().setContentType(JSON)
                .setBaseUri(baseUri)
                .setBasePath(basePath)
                .addFilter(new AllureLoggerFilter());
    }

    static LogConfig getLogConfig(boolean loggerEnabled) {
        if (loggerEnabled) {
            return logConfig().enableLoggingOfRequestAndResponseIfValidationFails();
        }
        return logConfig();
    }

    static RestAssuredConfig baseRestAssuredConfig() {
        return config().objectMapperConfig(objectMapperConfig().defaultObjectMapper(gson()))
                .failureConfig(failureConfig().failureListeners(new AllureFailureListener()));
    }
}

File: classfields/tests-core/src/main/scala/ru/auto/tests/ra/RequestSpecSupplier.scala
package ru.auto.tests.ra

import io.restassured.builder.RequestSpecBuilder
import io.restassured.config.LogConfig
import io.restassured.config.LogConfig.logConfig
import io.restassured.http.ContentType.JSON
import ru.auto.tests.commons.restassured.AllureLoggerFilter

import java.net.URI

object RequestSpecSupplier {

  def baseReqSpecBuilder(baseUri: URI): RequestSpecBuilder = {
    new RequestSpecBuilder()
      .setContentType(JSON)
      .setBaseUri(baseUri)
      .addFilter(new AllureLoggerFilter)
  }

  def getLogConfig(loggerEnabled: Boolean): LogConfig = {
    if (loggerEnabled) logConfig.enableLoggingOfRequestAndResponseIfValidationFails
    else logConfig
  }
}
